docs/source/starcluster/overview.rst

```diff
@@ -0,0 +1,197 @@
+the following information in your cluster configuration::
+
+  [cluster smallcluster]
+  ..
+  NODE_INSTANCE_TYPE=t2.micro NODE_IMAGE_ID=ami-6b211202
+  ..
+
+Now you can launch an EC2 instance using::
+
+  $ starcluster start smallcluster
+
+You can SSH into the master node by running::
+
+  $ starcluster sshmaster smallcluster
+
+You can transfer files to the nodes using the ``get`` and ``put`` commands as::
+
+  $ starcluster put /path/to/local/file/or/dir /remote/path/ $ starcluster get
+  /path/to/remote/file/or/dir /local/path/
+  
+Finally, you can terminate the instance by running::
+  
+  $ starcluster terminate smallcluster
+
+Setting up PySPH for StarCluster
+++++++++++++++++++++++++++++++++
+
+Most of the public AMIs currently distributed for StarCluster are outdated and
+have reached their end of life. To ensure a hassle-free experience while
+further extending the AMI and installing packages, you can use the 64 bit
+Ubuntu 16.04 AMI with AMI ID ``ami-01fdc27a`` which has most StarCluster
+dependencies and PySPH dependencies installed.
+
+Base AMI for PySPH [Optional]
+`````````````````````````````
+
+The ``ami.sh`` file which can be found in the ``starcluster`` directory in the
+PySPH repository automatically launches a vanilla 64-bit Ubuntu 16.04 instance,
+installs any necessary StarCluster and PySPH dependencies and saves an AMI with
+this configuration on your AWS account ::
+
+  $ ./ami.sh
+
+The AMI ID of the generated image is stored in ``AMI_ID``. You can also see a
+list of the AMIs currently in your AWS account by running ::
+
+  $ starcluster listimages
+
+Cluster configuration for PySPH
+```````````````````````````````
+
+Modify your StarCluster configuration file with the following
+information. Launching a cluster with the following configuration will start 2
+t2.micro instances, install the latest version of PySPH in each and keep track
+of the nodes loaded in ``/home/pysph/PYSPH_HOSTS``::
+   
+   [cluster pysphcluster]
+   KEYNAME = mykey CLUSTER_SIZE = 2 # Number of nodes in cluster
+   CLUSTER_USER = pysph
+   CLUSTER_SHELL = bash
+   NODE_IMAGE_ID = ami-01fdc27a # Or AMI ID for base AMI generated previously
+   NODE_INSTANCE_TYPE = t2.micro # EC2 Instance type PLUGINS = pysph_install
+   
+   [plugin pysph_install]
+   setup_class = pysph.PySPHInstaller
+
+Also, copy ``pysph.py`` from the ``starcluster`` directory to
+``~/.starcluster/plugins/``
+
+Running PySPH scripts on a cluster
+++++++++++++++++++++++++++++++++++
+
+You can start the cluster configured previously by running ::
+
+  $ starcluster start -c pysphcluster cluster
+
+Assuming your PySPH file ``cube.py`` is in the local home directory, you can
+first transfer this file to the cluster::
+
+  $ starcluster put -u pysph cluster ~/cube.py /home/pysph/cube.py
+
+Then run the PySPH code as::
+  
+  $ starcluster sshmsater -u pysph cluster "mpirun -n 2 --hostfile ~/PYSPH_HOSTS python ~/cube.py"
```

Comment:
No. The user doesn't need to do that. The `activate` script is automatically sourced from `/etc/profile.d/pysph.sh`. It's sourced from `~/.bashrc` for MPI processes running on other instances.